src/client/camera.cpp

```diff
@@ -694,13 +695,31 @@ void Camera::drawNametags()
 			i = m_nametags.begin();
 			i != m_nametags.end(); ++i) {
 		Nametag *nametag = *i;
+		core::vector3df cam_pos = m_cameranode->getAbsolutePosition();
+		core::vector3df nametag_pos = nametag->parent_node->getAbsolutePosition();
+
+		if (!nametag->nametag_viewthrough_walls) {
+			if (cam_pos.getDistanceFrom(nametag_pos) >
+					stoi(g_settings->get("viewing_range")) &&
```

Comment:
I think deciding on this condition alone is the best approach.
So, nametag is not drawn if beyond player 'view range'.
The PR implements a 'realistic' hiding of nametags by objects, so it makes sense that nametags are 'realistically' only visible within the player's view range.
The added benefit is that because view range is usually only 100-200 nodes, the raycasts are not excessively long.

Reply:
Alrighty, done